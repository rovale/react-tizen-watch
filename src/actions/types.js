export const INITIALIZE_APP = 'INITIALIZE_APP';

export const LOAD_PAGES = 'LOAD_PAGES';
export const LOAD_DEVICES = 'LOAD_DEVICES';
export const LOAD_GROUPS = 'LOAD_GROUPS';

export const ACTIVATE_PAGE = 'ACTIVATE_PAGE';
export const SELECT_PAGE = 'SELECT_PAGE';
export const CLOSE_PAGE = 'CLOSE_PAGE';

export const ACTIVATE_DEVICE = 'ACTIVATE_DEVICE';
export const SELECT_DEVICE = 'SELECT_DEVICE';
export const CLOSE_DEVICE = 'CLOSE_DEVICE';
export const CHANGE_DEVICE_ATTRIBUTE = 'CHANGE_DEVICE_ATTRIBUTE';
export const TOGGLE_SWITCH = 'TOGGLE_SWITCH';

export const ACTIVATE_BUTTON = 'ACTIVATE_BUTTON';
export const CLICK_BUTTON = 'CLICK_BUTTON';
