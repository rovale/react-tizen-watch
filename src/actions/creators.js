import * as actionType from './types';

export const initializeApp = () => ({ type: actionType.INITIALIZE_APP });
export const handleError = (msg, url, lineNo, columnNo, error) => ({ type: actionType.HANDLE_ERROR, payload: { msg, url, lineNo, columnNo, error } });
export const loadPages = pages => ({ type: actionType.LOAD_PAGES, payload: { pages } });
export const loadDevices = devices => ({ type: actionType.LOAD_DEVICES, payload: { devices } });

export const activatePage = id => ({ type: actionType.ACTIVATE_PAGE, payload: { id } });
export const selectPage = id => ({ type: actionType.SELECT_PAGE, payload: { id } });
export const closePage = () => ({ type: actionType.CLOSE_PAGE });

export const activateDevice = id => ({ type: actionType.ACTIVATE_DEVICE, payload: { id } });
export const selectDevice = id => ({ type: actionType.SELECT_DEVICE, payload: { id } });
export const closeDevice = () => ({ type: actionType.CLOSE_DEVICE });
export const changeDeviceAttribute = event => ({ type: actionType.CHANGE_DEVICE_ATTRIBUTE, payload: { event } });

export const toggleSwitch = id => ({ type: actionType.TOGGLE_SWITCH, payload: { id } });

export const activateButton = id => ({ type: actionType.ACTIVATE_BUTTON, payload: { id } });
export const clickButton = id => ({ type: actionType.CLICK_BUTTON, payload: { id } });
