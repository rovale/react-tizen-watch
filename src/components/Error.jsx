import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { Page, Header, Content } from './common/Page';

const Error = ({
  msg, url, lineNo, columnNo, error,
}) => (
  <Page>
    <Header>Error</Header>
    <Content>
      Msg: {msg}<br />
      Url: {url}<br />
      LineNo: {lineNo}<br />
      ColumnNo: {columnNo}<br />
      Error: {!!error && error.stack }
    </Content>
  </Page>
);

Error.propTypes = {
  msg: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  lineNo: PropTypes.number.isRequired,
  columnNo: PropTypes.number.isRequired,
  error: PropTypes.shape({
    stack: PropTypes.string.isRequired,
  }).isRequired,
};

const mapStateToProps = state => ({
  msg: state.ui.error.msg,
  url: state.ui.error.url,
  lineNo: state.ui.error.lineNo,
  columnNo: state.ui.error.columnNo,
  error: state.ui.error.error,
});

export default connect(mapStateToProps)(Error);
