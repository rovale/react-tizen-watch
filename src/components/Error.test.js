import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { createStore } from 'redux';
import reducer from '../reducers';
import * as action from '../actions/creators';

import Error from './Error';

Enzyme.configure({ adapter: new Adapter() });

test('it should render as expected', () => {
  const store = createStore(reducer);
  store.dispatch(action.handleError('someMsg', 'someUrl', 10, 20, { stack: 'someErrorStack' }));
  const wrapper = mount((<Error store={store} />)).children().first().children();
  expect(wrapper).toMatchSnapshot();
});
