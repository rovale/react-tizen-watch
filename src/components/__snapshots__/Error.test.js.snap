// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render as expected 1`] = `
<Page>
  <div
    className="ui-page ui-page-active ui-scroll-on"
  >
    <div
      className="ui-scroller ui-snap-container"
    >
      <Component>
        <header
          className="ui-header"
        >
          <h2
            className="ui-title"
          >
            Error
          </h2>
        </header>
      </Component>
      <Component>
        <div
          className="ui-content"
        >
          Msg: 
          someMsg
          <br />
          Url: 
          someUrl
          <br />
          LineNo: 
          10
          <br />
          ColumnNo: 
          20
          <br />
          Error: 
          someErrorStack
        </div>
      </Component>
    </div>
  </div>
</Page>
`;
